tensorflow/compiler/mlir/xla/experimental/conv_emitter/conv_emitter.cc

```diff
@@ -150,7 +150,7 @@ mlir::Operation* HoistAndFix(llvm::iplist<mlir::Operation>::iterator begin_op,
     ancestors.resize(i + 1);
```

Comment:
The decltype is really not something I'd like to read here when an int just works.

You could rewrite the entire block into something that looks like this though:

```
int lastValidAncestor = std::distance(ancestors.begin(),
  llvm::find_if(ancestors, [&] (mlir::AffineForOp for_op) {
    return for_op.getBody()->front() == &*begin_op;
  }));
  ancestors.resize(lastValidAncestor + 1);
```

Reply:
Is this code not correct?